packages/foundry/src/vault/strategy/autocompouding/liquidityPair/protocols/sushi/MiniChefV2Compounder.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: GPL-3.0
+// Docgen-SOLC: 0.8.15
+
+pragma solidity ^0.8.15;
+import { LiquidityPairBase, ERC20 } from "../../LiquidityPairBase.sol";
+import { IUniswapRouterV2 } from "../../../../../../interfaces/external/uni/IUniswapRouterV2.sol";
+import { IUniswapV2Pair } from "../../../../../../interfaces/external/uni/IUniswapV2Pair.sol";
+import { IMiniChefV2 } from "../../../../../../interfaces/external/sushi/IMiniChefV2.sol";
+import { IRewarder } from "../../../../../../interfaces/external/IRewarder.sol";
+
+contract LiquidityPairCompounder is LiquidityPairBase {
+  constructor(
```

Comment:
Id like to add a tradeModule and an lpModule to the constructor.
The rest of the contract would stay the same but instead of overriding the actual swap or LP deposit part we would delegate that to a very small contract which encapsulates this logic in a unified interface. 
This would allow us to keep this Compounder contract as is and simply slot in varying AMMs without needing to override this contract for all the possible AMM -> LP combinations.